File: python/aport_guardrails/frameworks/langchain.py
"""LangChain/LangGraph adapter: AsyncCallbackHandler."""

from aport_guardrails.frameworks.base import BaseAdapter


class LangChainAdapter(BaseAdapter):
    @property
    def name(self) -> str:
        return "langchain"

    async def detect(self) -> bool:
        return False

    async def install(self) -> None:
        pass

    async def verify(self) -> bool:
        return True

    async def test(self) -> bool:
        return True

File: packages/core/src/frameworks/langchain.ts
/**
 * LangChain/LangGraph adapter: AsyncCallbackHandler integration.
 * Reserved for programmatic use; CLI dispatch is bin/agent-guardrails (bash).
 */

import type { BaseAdapter } from './base.js';

export const langchainAdapter: BaseAdapter = {
  name: 'langchain',

  async detect() {
    return false;
  },

  async install() {
    // Run bin/frameworks/generic.sh (via bin/agent-guardrails --framework=langchain)
  },

  async verify() {
    return true;
  },

  async test() {
    return true;
  },
};
